src/Functions/FunctionsStringBytes.cpp

```diff
@@ -0,0 +1,142 @@
+#include <cmath>
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionsStringBytes.h>
+#include <Functions/IFunction.h>
+#include <Common/BitHelpers.h>
+#include <Common/PODArray.h>
+
+namespace DB
+{
+
+inline UInt8 countBits(UInt64 x)
+{
+#if defined(__clang__)
+    return __builtin_popcountll(x);
+#else
+    UInt8 count = 0;
+    while (x)
+    {
+        count += x & 1;
+        x >>= 1;
+    }
+    return count;
+#endif
+}
+
+class ByteCounters
+{
+private:
+    static constexpr size_t COUNTERS_SIZE = 256;
+    UInt32 counters[COUNTERS_SIZE] = {0};
+    UInt32 current_generation = 0;
```

Comment:
Btw, was this optimization worth it?

Reply:
I didn't benchmark it specifically, but it definitely seems to be worth it because instead of resetting all 256 counters (O(n)) between strings, it uses the bespoke generation counter, which is an O(1) operation. For columns with many sequential strings, this approach will be significantly faster due to lower cost of initialisation for the individual strings.